File: thrifts-dotnet/ThriftS.Serializer/MapContainer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThriftS.Common;
using ThriftS.Common.Attributes;

namespace ThriftS.Serializer
{
    [ThriftSModel]
    internal class MapContainer<TKey, TValue>
    {
        [ThriftSMember(1)]
        public Dictionary<TKey, TValue> Value { get; set; }

        public MapContainer()
        {
        }

        public MapContainer(Dictionary<TKey, TValue> value)
        {
            Value = value;
        }
    }
}

File: thrifts-java/thrifts-serializer/src/main/java/thrifts/serializer/MapContainer.java
package thrifts.serializer;

import thrifts.common.annotations.ThriftSMember;
import thrifts.common.annotations.ThriftSModel;

import java.util.HashMap;

@ThriftSModel
final class MapContainer<TKey, TValue> {
    @ThriftSMember(tag = 1)
    private HashMap<TKey, TValue> value;

    public MapContainer()
    {
    }

    public MapContainer(HashMap<TKey, TValue> value)
    {
        this.value = value;
    }

    public HashMap<TKey, TValue> getValue() {
        return this.value;
    }

    public void setValue(HashMap<TKey, TValue> value) {
        this.value = value;
    }
}
